diff --git a/kernel/supercall/supercall.c b/kernel/supercall/supercall.c
index eab7fdb08..9e2d6a38f 100644
--- a/kernel/supercall/supercall.c
+++ b/kernel/supercall/supercall.c
@@ -90,6 +90,8 @@ static inline int ksu_handle_fd_request(void __user *arg4)
 	return 0;
 }
 
+#define KSU_GET_LKM_VARIANT 20000
+
 // downstream: make sure to pass arg as reference, this can allow us to extend things.
 int ksu_handle_sys_reboot(int magic1, int magic2, unsigned int cmd, void __user **arg)
 {
@@ -112,6 +114,12 @@ int ksu_handle_sys_reboot(int magic1, int magic2, unsigned int cmd, void __user
 		return ksu_handle_fd_request(arg4);
 	}
 
+	if (magic2 == KSU_GET_LKM_VARIANT && is_manager()) {
+		static const char variant[] = "xxKSU";
+		copy_to_user(arg4, variant, sizeof(variant));
+		return 0;
+	}
+
 	// only root is allowed for these commands
 	if (current_uid().val != 0)
 		return 0;
